include/swift/AST/DiagnosticsSema.def

```diff
@@ -2988,8 +2988,8 @@ ERROR(invalid_objc_decl,none,
 ERROR(invalid_objc_swift_rooted_class,none,
       "only classes that inherit from NSObject can be declared @objc", ())
 ERROR(invalid_nonobjc_decl,none,
-      "only methods, initializers, properties and subscript declarations can "
-      "be declared @nonobjc", ())
+      "@nonobjc can only be used with members of classes, @objc protocols, and "
```

Comment:
A previous PR has added a specific error handling for the non class extensions marked as `@nonobj`, so is better to drop that part in this one? Maybe only tweaking the original message like "only methods, initializers, properties, and subscript declarations inside classes can be declared @nonobjc"